api/src/main/resources/descriptions/io.strimzi.api.kafka.model.authentication.KafkaClientAuthenticationOAuth.adoc

```diff
@@ -0,0 +1,93 @@
+To use OAuth client authentication, set the `type` property to the value `oauth`.
+OAuth authentication can be configured in several different ways.
+
+.Client ID and secret
+
+You can configure the address of your OAuth server in the `tokenEndpointUri` property together with the OAuth client ID and client secret.
+The OAuth client will connect to the OAuth server, authenticate using the client ID and secret and get and access token which it will use to authenticate with the Kafka broker.
+In the `clientSecret` property, specify a link to a `Secret` containing the client secret.
+
+.An example of OAuth client authentication using client ID and client secret
+[source,yaml,subs=attributes+]
+----
+authentication:
+  type: oauth
+  tokenEndpointUri: https://sso.myproject.svc:8443/auth/realms/internal/protocol/openid-connect/token
+  clientId: my-client-id
+  clientSecret:
+    secretName: my-client-oauth-secret
+    key: client-secret
+----
+
+.Client ID and refresh token
+
+You can configure the address of your OAuth server in the `tokenEndpointUri` property together with the OAuth client ID and refresh token.
+The OAuth client will connect to the OAuth server, authenticate using the client ID and refresh token and get and access token which it will use to authenticate with the Kafka broker.
```

Comment:
```suggestion
The OAuth client will connect to the OAuth server, authenticate using the client ID and refresh token and get an access token which it will use to authenticate with the Kafka broker.
```